src/rgw/rgw_op.h

```diff
@@ -2029,8 +2029,37 @@ class RGWGetHealthCheck : public RGWOp {
 
 
 class RGWDeleteMultiObj : public RGWOp {
+  /**
+   * Handles the deletion of an individual object and uses
+   * set_partial_response to record the outcome. 
+   */
+  void handle_individual_object(const rgw_obj_key *o, optional_yield y);
+  
+  /**
+   * When the request is being executed in a coroutine, performs
+   * the actual formatter flushing and is responsible for the
+   * termination condition (when when all partial object responses
+   * have been sent). Note that the formatter flushing must be handled
+   * on the coroutine that invokes the execute method vs. the 
+   * coroutines that are spawned to handle individual objects because
+   * the flush logic uses a yield context that was captured
+   * and saved on the req_state vs. one that is passed on the stack.
+   * This is a no-op in the case where we're not executing as a coroutine.
+   */
+  void wait_flush(optional_yield y, std::function<bool()> predicate);
+
 protected:
   std::vector<delete_multi_obj_entry> ops_log_entries;
+
+  /**
+   * Acts as an async condition variable when the request is being
+   * executed on a coroutine. Formatter flushing must happen on the main
+   * request coroutine vs. spawned coroutines, so spawned coroutines use
+   * the cancellation of this timer to notify the main coroutine when
+   * data is ready to flush. 
+   */
+  std::unique_ptr<boost::asio::deadline_timer> formatter_flush_cond;
```

Comment:
instead of being a data member, could this be a local variable in the `execute()` function, and passed as an argument to the called functions?

Reply:
yeah I can move it :+1: